modules/typesniffer/typesniffer.go

```diff
@@ -97,6 +98,15 @@ func DetectContentType(data []byte) SniffedType {
 		ct = SvgMimeType
 	}
 
+	if strings.HasPrefix(ct, "audio/") && bytes.HasPrefix(data, []byte("ID3")) {
+		// the MP3 detection is quite inaccurate, any content with "ID3" prefix will result in "audio/mpeg"
+		// so remove the "ID3" prefix and detect again, if result is text, then it must be text content.
+		ct2 := http.DetectContentType(data[3:])
```

Comment:
Hmm…
I wonder how large the amount of valid audio files will be that are now detected as a text file.

What we could do is merge this PR with reservations, and if it turns out to be too breaking for audio files to revert it.

My current thinking comes from pure probability theory:
Is the probability of false negatives with this PR smaller than the probability of false positives without it?

Reply:
It won't break, because a valid MP3 file has many `\000` and other non-printable bytes in it, so `http.DetectContentType` won't make mistakes on them.

Reply:
```suggestion
		// This is especially because audio files contain many unprintable/invalid characters like "\000" 
		ct2 := http.DetectContentType(data[3:])
```